scripts/data_migration/migrate_assessment_subject_day_data.js

```diff
@@ -0,0 +1,80 @@
+const { MongoClient } = require('mongodb')
+
+const COLLECTION_NAME = 'assessmentSubjectDayData'
+
+export async function migrateAssessmentSubjectDayData() {  
+  const mongoConnection = await MongoClient.connect(process.env.MONGODB_URI, {
+    ssl: process.env.NODE_ENV === 'production',
+    useNewUrlParser: true,
+  })
+
+  const appDb = mongoConnection.db('dpdmongo')
+  const db = mongoConnection.db('dpdata')
+  const tocCollection = db.collection('toc')
+
+  const collectionSet = new Set()
+  const tocCursor = await tocCollection.find({})
+
+  while (await tocCursor.hasNext()) {
+    const assessmentSubjectDays = []
+    const toc = await tocCursor.next()
+    
+    toc['toc_id'] = toc['_id']
+    delete toc['_id']
+    toc['toc_mtime'] = toc['mtime']
+    delete toc['mtime']
+    toc['toc_path'] = toc['path']
+    delete toc['path']
+
+    if (collectionSet.has(toc.collection)) {
+      continue
+    }
+
+    const dayData = await db.collection(toc.collection).find({}).toArray()
+    collectionSet.add(toc.collection)
+
+    new Set(dayData.map( d => Object.keys(d)).reduce((acc, keys) => acc.concat(keys), [])).forEach((key) => {
+      if (toc[key] !== undefined  && dayData[0][key] !== toc[key]) {
+        mongoConnection.close()
+        throw new Error(`Prevented write due to overwritten data in ${toc.collection}: ${key}`)
+      }
+    })
+    
+    for (const day of dayData) {
+      day['legacy_id'] = day['_id']
+      delete day['_id']
+
+      const assessmentSubjectDay = {
+        ...toc,
```

Comment:
I'm conflicted, the toc has file data that is outside of the what the assessment day data is, perhaps it is best to contain the toc collection for the csv file, what do you think..?

Reply:
I get it, I think we can do better on this data structure. I think it would be better to discuss in a huddle/meet though, does that make sense to you?

Reply:
yeah that makes sense to me

Reply:
From our conversation:

Reply:
[51473fa](https://github.com/AMP-SCZ/dpdash/pull/386/commits/51473fa5a272e896ee64e1eedec118c833df5d9d)